### PR TITLE
Core/PacketsIO: Updated and enabled SMSG_CHAR_CUSTOMIZE, SMSG_CHAR_CU…

### PR DESCRIPTION
**Pull request by DJScias**
…STOMIZE_FAILED

### CODE REVIEW

**Comment by DJScias** [comment]
Fixing something in regards to @Carbenium's comment.

File: src/server/game/Handlers/CharacterHandler.cpp [hunk 1]
```diff
@@ -2574,25 +2574,29 @@ void WorldSession::SendCharRename(ResponseCodes result, WorldPackets::Character:
 
 void WorldSession::SendCharCustomize(ResponseCodes result, WorldPackets::Character::CharCustomizeInfo const* customizeInfo)
 {
-    /// @todo: fix 6.x implementation
-    (void)result;
-    (void)customizeInfo;
-    /*
-    WorldPacket data(SMSG_CHAR_CUSTOMIZE, 1 + 8 + customizeInfo.NewName.size() + 1 + 6);
-    data << uint8(result);
+    WorldPackets::Character::CharCustomizeResponse response;
```
Comment (joschiwald) [inline src/server/game/Handlers/CharacterHandler.cpp]: move this line inside the if block

Comment (DJScias) [inline src/server/game/Handlers/CharacterHandler.cpp]: Will do, just a moment.

**Comment by DJScias** [comment]
@joschiwald Done.

File: src/server/game/Server/Packets/CharacterPackets.cpp [hunk 2]
```diff
@@ -498,3 +498,26 @@ WorldPacket const* WorldPackets::Character::SetFactionVisible::Write()
...
+WorldPacket const* WorldPackets::Character::CharCustomizeResponse::Write()
+{
+    _worldPacket << Guid;
+    _worldPacket << uint8(SexID);
+    _worldPacket << uint8(SkinID);
+    _worldPacket << uint8(HairColorID);
+    _worldPacket << uint8(HairStyleID);
+    _worldPacket << uint8(FacialHairStyleID);
+    _worldPacket << uint8(FaceID);
+    _worldPacket.WriteBits(Name.length(), 6);
```
Comment (joschiwald) [inline src/server/game/Server/Packets/CharacterPackets.cpp]: missing flush

Comment (Warpten) [inline src/server/game/Server/Packets/CharacterPackets.cpp]: IIRC WriteString flushes. for obvious reasons.

Comment (joschiwald) [inline src/server/game/Server/Packets/CharacterPackets.cpp]: not if the string is empty

File: src/server/game/Server/Packets/CharacterPackets.h [hunk 3]
```diff
@@ -686,6 +686,34 @@ namespace WorldPackets
 
             uint32 FactionIndex = 0;
         };
+
+        class CharCustomizeResponse final : public ServerPacket
+        {
+        public:
+            CharCustomizeResponse() : ServerPacket(SMSG_CHAR_CUSTOMIZE, 16 + 1 + 4 + 4 + 4 + 4 + 4 + 4) { }
```
Comment (jackpoz) [inline src/server/game/Server/Packets/CharacterPackets.h]: what's the math behind this packet size ?
```
16 + 1 + 4 + 4 + 4 + 4 + 4 + 4
```

Comment (Robens) [inline src/server/game/Server/Packets/CharacterPackets.h]: prob all the 4 should be 1 instead, same for SMSG_CHAR_CUSTOMIZE_FAILED

Comment (DJScias) [inline src/server/game/Server/Packets/CharacterPackets.h]: Yes, I did a miss-calculation there. Just got home, so I'll fix it soon.

File: src/server/game/Server/Packets/CharacterPackets.cpp (hunk 2 above)
Comment (DJScias) [inline src/server/game/Server/Packets/CharacterPackets.cpp]: Flushed the bits.

File: src/server/game/Server/Packets/CharacterPackets.cpp [hunk 4]
```diff
@@ -498,3 +498,27 @@ WorldPacket const* WorldPackets::Character::SetFactionVisible::Write()
     _worldPacket << FactionIndex;
     return &_worldPacket;
 }
+
+WorldPacket const* WorldPackets::Character::CharCustomizeResponse::Write()
+{
+    _worldPacket << Guid;
```
Comment (Carbenium) [inline src/server/game/Server/Packets/CharacterPackets.cpp]: CharGUID

**Pull request closed without merging**